modules/dnn/src/dnn_common.hpp

```diff
@@ -15,6 +15,17 @@ void initializeLayerFactory();
 
 namespace detail {
 
+class NotImplemented : public Layer
+{
+public:
+    static Ptr<Layer> create(const LayerParams &params);
+
+    static void Register();
+    static void unRegister();
+};
+
+Ptr<Layer> notImplementedRegisterer(LayerParams &params);
+
```

Comment:
we don't need that in `dnn_common.hpp` anymore.

Implementation can be placed near `NotImplemented::Register()`

Reply:
It is used in TFimporter::parseNode.

Reply:
Why is not `NotImplemented::Register()`? (or nothing because it is handled in `enableModelDiagnostics()`)

Reply:
Let's imagine TFImporter doesn't know some layer. It will be registered in order to not flood the output with errors about it.

Reply:
If diagnostic mode is not enabled then strategy must be "fail fast". We should not eat problems.

Reply:
I am indeed talking about the behavior of this function with diagnostic mode turned on.

Reply:
Got it. It is registered under a **new name**.

There is another problem:
- "NotImplemented" is unregistered when we call `enableModelDiagnostics(false);`
- these unknown layers don't.

Perhaps we should disallow turning off diagnostic mode if it is already enabled. This would be OK for tool, but this would cause problems with tests.

---

BTW, `NotImplemented::Register()` can accept the "name" string parameter.

Reply:
The last problem can be resolved through singleton for registered `NotImplemented` layers. It should provide two methods:
- `registerLayerType(const std::string& name)` - register and save used name
- `unregisterLayerTypes()` (called by `enableModelDiagnostics(false)`) - cleanup all previously registered layer's types saved on registration

Reply:
I will push changes in a few hours. I added LayerHandler class which manages these layers and doesn't need to call registerLayer.

Reply:
Resolved.